### PR TITLE
All-benchmarks script and CI workflow improvements

### LINKED ISSUE #759: Integrate new benchmarks and improve benchmarks CI
- [x] storage
- [x] overheads -> https://github.com/Manta-Network/Manta/issues/760
- [x] machine
- [x] single CI job

### PR DESCRIPTION
**Pull request by ghzlatarev**
Signed-off-by: Georgi Zlatarev <georgi.zlatarev@manta.network>

## Description

closes: #759 

* Adds a `run_all_benchmarks.sh` script which runs pallet, storage and machine benchmark and outputs results, same as this Parity script https://github.com/paritytech/substrate/pull/11493
* Adds new CI workflow which calls the script and replaces the 3 duplicated weights benchmarking workflows for dolphin, calamari and manta.
* The chain-spec is a mandatory input field on the new workflow.
* The storage benchmark can be skipped if no URL is provided for the second input field that specifies a tar.gz archive of a production chain snapshot.
* Changed all CI workflows to always try to upload artifacts even if previous steps have failed, which is useful for debugging failed jobs.
* Disable parachain block parsing until polkadot-js recognizes we that our chain is a special case like Moonbeam with nimbus keys.
* Update release process template to account for aforementioned changes.

* Example run - https://github.com/Manta-Network/Manta/runs/7880566445?check_suite_focus=true

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Added **one** label out of the `L-` group to this PR
- [x] Added **one or more** labels out of `A-` and `C-` groups to this PR
- [x] Re-reviewed `Files changed` in the Github PR explorer.

Situational Notes:
- If adding functionality, write unit tests!
- If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.
- If runtime changes, need to update the version numbers properly:
   * `authoring_version`: The version of the authorship interface. An authoring node will not attempt to author blocks unless this is equal to its native runtime.
   * `spec_version`: The version of the runtime specification. A full node will not attempt to use its native runtime in substitute for the on-chain Wasm runtime unless all of spec_name, spec_version, and authoring_version are the same between Wasm and native.
   * `impl_version`: The version of the implementation of the specification. Nodes are free to ignore this; it serves only as an indication that the code is different; as long as the other two versions are the same then while the actual code may be different, it is nonetheless required to do the same thing. Non-consensus-breaking optimizations are about the only changes that could be made which would result in only the impl_version changing.
   * `transaction_version`: The version of the extrinsics interface. This number must be updated in the following circumstances: extrinsic parameters (number, order, or types) have been changed; extrinsics or pallets have been removed; or the pallet order in the construct_runtime! macro or extrinsic order in a pallet has been changed. You can run the `metadata_diff.yml` workflow for help. If this number is updated, then the `spec_version` must also be updated
- Verify benchmarks & weights have been updated for any modified runtime logics

### CODE REVIEW

**Review by Apokalip — DISMISSED** [review]
Looks pretty good to me! Now it looks a little cleaner than before and the tests' always() is a good change as we would want to see not only until the failing test

**Review by Garandor — CHANGES_REQUESTED** [review]
Missing a C- label

**Comment by Apokalip** [comment]
> Missing a C- label

@Garandor 
We should probably add the other labels in the label checker workflow? I keep forgetting about them too

File: .github/workflows/publish_draft_releases.yml [hunk 1]
```diff
@@ -550,26 +569,32 @@ jobs:
...
         with:
           name: block-time-rococo-relay-for-${{ matrix.chain-spec.id }}.json
           path: ${{ github.workspace }}/block-time-rococo.json
-      - name: parse calamari block times
-        run: |
-          grep '#.*' ${{ github.workspace }}/measure-block-time-${{ matrix.chain-spec.id }}-stdout.log | while read -r line; do words=($line); echo ${words[6]},${words[10]} | tee ${{ github.workspace }}/block-time-${{ matrix.chain-spec.id }}.csv; done
-          if [ ! -f ${{ github.workspace }}/block-time-${{ matrix.chain-spec.id }}.csv ]; then echo "block times not detected"; exit 1; fi
-          jq -s -R '[split("\n") | .[] | select(length > 0) | split(",") | {block:.[0]|tonumber, time:.[1]|tonumber} ]' ${{ github.workspace }}/block-time-${{ matrix.chain-spec.id }}.csv > ${{ github.workspace }}/block-time-${{ matrix.chain-spec.id }}.json
-      - uses: actions/upload-artifact@v2
+# - if: always()
```
Comment (Garandor) [inline .github/workflows/publish_draft_releases.yml]: pls indent comment if possible

**Review by Garandor — DISMISSED** [review]
LGTM, maybe get rid of the yamllint warning before merge @ghzlatarev

**Review by Dengjianping — DISMISSED** [review]
LGTM, but there's conflict to be resolved.

File: .github/workflows/publish_draft_releases.yml (hunk 1 above)
Comment (ghzlatarev) [reply .github/workflows/publish_draft_releases.yml]: It bugs the workflow for some reason.

Comment (ghzlatarev) [reply .github/workflows/publish_draft_releases.yml]: Nvm, worked here, and cleared the warning.

**Review by Garandor — APPROVED** [review]

**Review by Dengjianping — APPROVED** [review]

**Pull request merged**